gui/src/containers/Jobs/Definition.js

```diff
@@ -0,0 +1,91 @@
+import React, { Component } from 'react'
+import { withStyles } from '@material-ui/core/styles'
+import Grid from '@material-ui/core/Grid'
+import Card from '@material-ui/core/Card'
+import CardContent from '@material-ui/core/CardContent'
+import Typography from '@material-ui/core/Typography'
+import Divider from '@material-ui/core/Divider'
+import Content from 'components/Content'
+import RegionalNav from 'components/Jobs/RegionalNav'
+import PrettyJson from 'components/PrettyJson'
+import { connect } from 'react-redux'
+import { fetchJob, createJobRun } from 'actions'
+import jobSelector from 'selectors/job'
+import matchRouteAndMapDispatchToProps from 'utils/matchRouteAndMapDispatchToProps'
+import jobSpecDefinition from 'utils/jobSpecDefinition'
+
+const styles = theme => ({
+  definitionTitle: {
+    marginTop: theme.spacing.unit * 2,
+    marginBottom: theme.spacing.unit * 2
+  },
+  divider: {
+    marginTop: theme.spacing.unit,
+    marginBottom: theme.spacing.unit * 3
+  }
+})
+
+const renderDetails = ({job, classes}) => {
+  const definition = job && jobSpecDefinition(job)
+
+  if (definition) {
+    return (
+      <Grid container>
+        <Grid item xs={12}>
+          <Typography variant='title' className={classes.definitionTitle}>
+            Definition
+          </Typography>
+        </Grid>
+        <Grid item xs={12}>
+          <Divider light className={classes.divider} />
+        </Grid>
+        <Grid item xs={12}>
+          <PrettyJson object={definition} />
+        </Grid>
+      </Grid>
+    )
+  }
+
+  return <React.Fragment>Fetching ...</React.Fragment>
+}
+
+class Definition extends Component {
+  componentDidMount () {
```

Comment:
I feel like it's time we experiment with hooks and `useEffect`: https://reactjs.org/docs/hooks-intro.html

Reply:
Agreed we should experiment with hooks but they don't drop until `16.7.x`

Reply:
I've converted this and other classes to functions [with this](https://github.com/smartcontractkit/chainlink/pull/770/files?w=1#diff-c61189087b3cd7465b67f88d6a7450a7)

Once hooks drops in 16.7.x the package "use-react-hooks": "^1.0.7" I added can be removed and all the imports will be from `'react'` instead to finish the proper canonical upgrade.